neo4j/example/src/test/java/example/springdata/neo4j/ActorRepositoryIntegrationTest.java

```diff
@@ -57,7 +60,54 @@ public void shouldBeAbleToSaveAndLoadActor() {
 		assertThat(actorRepository.findById(daniel.getId())).hasValueSatisfying(actor -> {
 			assertThat(actor.getName()).isEqualTo(daniel.getName());
 			assertThat(actor.getRoles()).hasSize(1).first()
-					.satisfies(role -> assertThat(role.getRole()).isEqualTo("Harry Potter"));
+				.satisfies(role -> assertThat(role.getRole()).isEqualTo("Harry Potter"));
 		});
 	}
+
+	@Test // SEE #386
+	public void shouldBeAbleToHandleNestedProperties() {
+
+		Assume.assumeTrue(thatSupportForNestedPropertiesIsAvailable());
+
+		Movie theParentTrap = new Movie("The Parent Trap");
+		Movie iKnowWhoKilledMe = new Movie("I Know Who Killed Me");
+
+		Actor lindsayLohan = new Actor("Lindsay Lohan");
+
+		lindsayLohan.actedIn(theParentTrap, "Hallie Parker");
+		lindsayLohan.actedIn(theParentTrap, "Annie James");
+		lindsayLohan.actedIn(iKnowWhoKilledMe, "Aubrey Fleming");
+		lindsayLohan.actedIn(iKnowWhoKilledMe, "Dakota Moss");
+		actorRepository.save(lindsayLohan);
+
+		Actor nealMcDonough = new Actor("Neal McDonough");
+		nealMcDonough.actedIn(iKnowWhoKilledMe, "Daniel Fleming");
+		actorRepository.save(nealMcDonough);
+
+		assertThat(actorRepository.findAllByRolesMovieTitle(iKnowWhoKilledMe.getTitle())).hasSize(2)
+			.extracting(Actor::getName).contains(lindsayLohan.getName(), nealMcDonough.getName());
+	}
+
+	private static boolean thatSupportForNestedPropertiesIsAvailable() {
```

Comment:
Spring Data has a `Version` class that could be used to check for valid ranges.

Reply:
Nice, thanks.